src/briefcase/integrations/subprocess.py

```diff
@@ -88,6 +89,46 @@ def get_process_id_by_command(
     return None
 
 
+def pause_console_control_if(sub_method):
```

Comment:
Not sure I like the naming here - this "reads" like a dangling clause: Pause console control if... what? The dangling "if" part would make sense if the decorator took an argument that was evaluated (e.g., `@pause_console_control_if(is_windows_batch_file)`), but I'm not sure that sort of refactoring is worth it.

Not sure I have an obvious alternative - maybe something like `@ensure_console_is_safe`?

Reply:
That's fair. This name was wonky. Updated it to `ensure_console_is_safe`.